lib/application.js

```diff
@@ -80,36 +87,46 @@ class Application extends events.EventEmitter {
     }
   }
 
-  async loadScript(place, fileName) {
-    const { name, ext } = path.parse(fileName);
-    if (ext !== '.js' || name.startsWith('.')) return;
+  async loadDomain(fileName) {
+    const rel = fileName.substring(this.domainPath.length + 1);
+    const [name, ext] = rel.split('.');
+    if (ext !== 'js' || name.startsWith('.')) return;
     const script = await this.createScript(fileName);
-    const scripts = this[place];
-    if (!script) {
-      scripts.delete(name);
-      return;
-    }
-    if (place === 'domain') {
-      const config = this.config.sections[name];
-      const sandbox = this.domainSandbox;
-      sandbox.application[name] = { config };
-      const exp = script.runInContext(sandbox, SCRIPT_OPTIONS);
-      if (config) exp.config = config;
-      this.sandbox.application[name] = exp;
-      this.sandboxInject(name, exp);
-      if (exp.start) exp.start();
-    } else {
-      scripts.set(name, script);
-    }
+    if (!script) return;
```

Comment:
Wouldn't it be better to throw/reject here (or in `createScript`) to have some sort of indication that something went wrong instead o silently finishing?

Reply:
We write logs in `Application.prototype.createScript` and return null, so if script is null we just do not need to load it